docs/flamework/additional-modules/components/configuration.md

```diff
@@ -0,0 +1,71 @@
+---
+title: Configuration
+---
+Components have a few features we haven't covered yet!
+
+## Predicate
+You can add a `predicate` function to a component to determine if the component should be constructed for a provided instance.
+
+This only affects Components created via **CollectionService**!
+```ts
+@Component({
+    tag: "Example",
+    predicate: ( instance ) => instance.FindFirstAncestorOfClass("PlayerGui") !== undefined,
+})
+export class ExampleComponent extends BaseComponent<> implements OnStart {}
+```
+
+## Instance Guard
+The `instanceGuard` is similar to `predicate`, but serves a slightly different purpose.
```

Comment:
I think it should also be noted that `instanceGuard` acts as a replacement to the automatically generated guard for the Instance type provided to `BaseComponent`